src/dir_walker.rs

```diff
@@ -28,15 +33,21 @@ pub struct WalkData<'a> {
     pub ignore_hidden: bool,
 }
 
-pub fn walk_it(dirs: HashSet<PathBuf>, walk_data: WalkData) -> (Vec<Node>, bool) {
+pub fn walk_it(
+    dirs: HashSet<PathBuf>,
+    walk_data: WalkData,
+    info_data: Arc<PAtomicInfo>,
+    info_conf: Arc<PConfig>,
```

Comment:
I think the info_* variables should be added to WalkData struct (and probably made optional as I'd like a flag to turn off the in-progress message [although it should default to ON] )